src/navigation/navigation.cc

```diff
@@ -395,34 +418,41 @@ PathOption Navigation::getGreedyPath(Vector2f goal_loc)
 		predictCollisions(path);
 		trimPathLength(path, goal_loc);
 		calculateClearance(path);
+		float clearance_padded = path.clearance - (car_width_/2+padding_*2);
+		if (clearance_padded < 0) clearance_padded = 1e-5;
 
 		path.distance_to_goal = (path.obstruction - goal_loc).norm(); // approximately
 
 		max_free_path_length = std::max(path.free_path_length, max_free_path_length);
-		max_clearance = std::max(max_clearance, path.clearance);
-		min_distance_to_goal = std::min(min_distance_to_goal, path.distance_to_goal);
+		max_clearance_padded = std::max(clearance_padded, max_clearance_padded);
+		min_distance_to_goal = std::min(path.distance_to_goal, min_distance_to_goal);
 
-		FPL_vec.push_back(path.free_path_length);
-		clearance_vec.push_back(path.clearance);
+		free_path_length_vec.push_back(path.free_path_length);
+		clearance_padded_vec.push_back(clearance_padded);
 		distance_to_goal_vec.push_back(path.distance_to_goal);
 	}
 
 	// Iterate through paths to find the best one
 	for (int i = 0; i < num_paths; i++)
 	{
-		float free_path_length 	= FPL_vec.at(i);
-		float clearance 		= clearance_vec.at(i);
-		float distance_to_goal 	= distance_to_goal_vec.at(i);
-		
-		// Calculate cost
-		float cost = - ( free_path_length / max_free_path_length ) * free_path_length_weight_	// (-) decrease cost with large FPL
-					 - ( clearance / max_clearance	)		   * clearance_weight_				// (-) decrease cost with large clearance
-					 + ( distance_to_goal / min_distance_to_goal ) * distance_to_goal_weight_;	// (+) increase cost with large distance to goal
-
-		// Evaluate Minimum Cost
-		if (cost < min_cost) {min_cost = cost; BestPath = PossiblePaths_.at(i);}
+		float free_path_length = free_path_length_vec.at(i);
+		float clearance_padded = clearance_padded_vec.at(i);
+		float distance_to_goal = distance_to_goal_vec.at(i);
```

Comment:
The way all three of these variable names line up is kinda amazing, not gonna lie

Reply:
haha it's a thing of beauty